### PR TITLE
Adding example of a streamer looping over a set of media

### PR DESCRIPTION
**Pull request by nipierre**
- Example showing how to do a never-ending stream looping over a user-defined set of media
- Can choose the chunk size

### CODE REVIEW

File: examples/streamer.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        )
+        .arg(
+            Arg::with_name("chunk-size")
+                .short("c")
+                .help("Chunk size")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let port = matches.value_of("port").unwrap_or("3333").to_owned();
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:38]: use [default_value()](https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.default_value) instead

File: examples/streamer.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    print!("Value for port: {}", port);
+
+    let url = format!("127.0.0.1:{}", port);
+    let mut srt_socket = SrtSocketBuilder::new_connect(&url).connect().await.unwrap();
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:42]: replace unwrap with `except("message here")`

File: examples/streamer.rs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    let chunk_size = matches
+        .value_of("chunk-size")
+        .unwrap_or("10")
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:46]: same here, use `default_value`

Comment (MarcAntoine-Arnaud) [reply examples/streamer.rs:46]: change default value to 1316, 10 bits is very small ...

File: examples/streamer.rs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        .unwrap();
+
+    print!("Stream hooked");
+
+    let files: Vec<_> = matches.values_of("files").unwrap().collect();
+
+    print!("{:?}", files);
+
+    loop {
+        for file in files.iter() {
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:58]:
```rust
for file in &files {
}
```

File: examples/streamer.rs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    let chunk_size = matches
+        .value_of("chunk-size")
+        .unwrap_or("10")
+        .to_owned()
+        .parse::<u64>()
+        .unwrap();
+
+    print!("Stream hooked");
+
+    let files: Vec<_> = matches.values_of("files").unwrap().collect();
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:53]: remove `: Vec<_>`

File: examples/streamer.rs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+    print!("{:?}", files);
+
+    loop {
+        for file in files.iter() {
+            print!("Open file {}", file);
+            let mut f = File::open(file)?;
+            let metadata = f.metadata()?;
+            let mut buffer = vec![0; chunk_size as usize];
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:62]: set `usize` at the creation of the variable

File: examples/streamer.rs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+    let files: Vec<_> = matches.values_of("files").unwrap().collect();
+
+    print!("{:?}", files);
+
+    loop {
+        for file in files.iter() {
+            print!("Open file {}", file);
+            let mut f = File::open(file)?;
+            let metadata = f.metadata()?;
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:61]: not needed

File: examples/streamer.rs [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+            let mut buffer = vec![0; chunk_size as usize];
+
+            for _ in 0..metadata.len() / chunk_size {
+                f.read_exact(&mut buffer)?;
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:65]: use `read()` and get the number of bytes readed.
If it's [0](https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read), it's EOF.

File: examples/streamer.rs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+                srt_socket
+                    .send((Instant::now(), Bytes::copy_from_slice(&buffer)))
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:67]: use `Bytes::from(&buffer)` no copy needed.

File: examples/streamer.rs [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+                    .await
+                    .unwrap();
```
Comment (MarcAntoine-Arnaud) [inline examples/streamer.rs:69]: use `?` to throw the error

**Comment by russelltg** [comment]
Wow---thanks both of you!

This does have the same problem that @MarcAntoine-Arnaud's examples had--it uses the [`send`](https://docs.rs/futures/0.3.5/futures/sink/trait.SinkExt.html#method.send) function on `Sink`, which has the disclaimer in the docs:

> Note that, because of the flushing requirement, it is usually better to batch together items to send via send_all, rather than flushing between each item.

I'd probably use `futures::stream::unfold` to create a stream.

File: examples/streamer.rs [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+            let mut f = File::open(file)?;
+
+            loop {
+                let mut buffer = vec![0; chunk_size];
+                let size = f.read(&mut buffer)?;
+                if size == 0 {
+                    break;
+                }
+                srt_socket
+                    .send((Instant::now(), Bytes::from(buffer)))
```
Comment (russelltg) [inline examples/streamer.rs:70]: Use send_all with a stream

**Comment by russelltg** [comment]
I just did some restructuring, this should be moved into srt-tokio/examples

**Pull request closed without merging**